### PR TITLE
Implement API Interactions for Leaderboard

### PR DESCRIPTION
**Pull request by CVILLA09**
## Description

In this pull request, I've implemented the functionality to interact with the Leaderboard API. 

The main changes include:

1. Creation of a new game using the API and storing the returned game ID.
2. Implementation of the "Refresh" button to fetch all scores for the created game from the API.
3. Implementation of the "Submit" button to send a new score for the created game to the API.

Key files modified:
- `leaderBoard.js`: Added methods to interact with the API.
- `index.js`: Added event listeners for the "Refresh" and "Submit" buttons.

This pull request closes the feature to hit the API for the leaderboard project.

### CODE REVIEW

**Review by Strangeal — APPROVED** [review]
mistake

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @CVILLA09, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text


## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Well structured project ✅ 

## Required Changes ♻️
 - [ ] Great job so far, however, I was unable to start your project because your `webpack` config hasn't been set up properly. Due to that, I was able to test the functionality of your code in this milestone. Kindly setup your `webpack` properly and make sure to test it before resubmitting your project for review. To learn how to set up webpack check out this [link](https://github.com/CVILLA09/Leaderboard/blob/1a5b9434e87e99f9df39390ced354e831bf2681b/webpack.config.js#L1-L9)
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by CVILLA09** [comment]
@Strangeal  



Hello Justice Bajeri,

Thank you for the feedback! I'm currently addressing the required changes, but I need more details about the Webpack configuration issue you mentioned. Could you please specify what's wrong with the configuration? Additionally, I'm unsure if the provided link for setting up Webpack is correct.

Your assistance is much appreciated!

Best regards,
Cristian Villa

File: webpack.config.js [hunk 1]
```diff
@@ -6,4 +6,4 @@ module.exports = {
     filename: 'main.js',
     path: path.resolve(__dirname, 'dist'),
   },
-};
```
Comment (bobb-Rob) [inline webpack.config.js]: - [ ] Thank you for your effort in setting up webpack for your project. As the previous reviewer mentioned, it seems there are some issues with the current webpack configuration. To help you get it right, I've provided an example of a properly set up webpack configuration below:

```javascript
const path = require('path');
const HtmlWebpackPlugin = require('html-webpack-plugin');

module.exports = {
  mode: 'development',
  entry: {
    bundle: path.resolve(__dirname, 'src/index.js'),
  },
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: '[name].js',
    clean: true,
    assetModuleFilename: '[name][ext]',
  },
  devtool: 'source-map',
  devServer: {
    static: {
      directory: path.resolve(__dirname, 'dist'),
    },
    port: 9000,
    open: true,
    hot: true,
    compress: true,
    historyApiFallback: true,
  },
  module: {
    rules: [
      {
        test: /\.css$/,
        use: [
          'style-loader',
          'css-loader',
        ],
      },
      {
        test: /\.js$/,
        exclude: /node_modules/,
        use: {
          loader: 'babel-loader',
          options: {
            presets: ['@babel/preset-env'],
          },
        },
      },
    ],
  },
  plugins: [
    new HtmlWebpackPlugin({
      title: 'Leaderboard App',
      filename: 'index.html',
      template: 'src/index.html',
    }),
  ],
};
```

Please use this webpack configuration as a guide and ensure it matches the setup provided on your student dashboard. Properly configured webpack is crucial for the smooth functioning of your project.

If you encounter any issues or have any questions during the setup process, feel free to ask for assistance. Happy coding!

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️

Hi @CVILLA09 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: webpack.config.js (hunk 1 above)
Comment (CVILLA09) [reply webpack.config.js]: @bobb-Rob

**Review by LionRouge1 — APPROVED** [review]
Hi @CVILLA09, 

# STATUS: APPROVED! 🥇 💯 
Great work on making the changes 👏 
You've done well implementing the requested changes

done

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

## Highlights
- Great job, your project is error free :+1: 
- Nice work you meet all the requirements :1st_place_medal: 
- All linters passed :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**